### DISCUSSION TITLE
Request: Make TaskList more dynamic

### DISCUSSION THREAD

**Discussion by mcnuggetsx20** [qtile/qtile#4297] [Ideas]
The current state of TaskList is a little weird. It can only take a fixed amount of space, that is not configurable. It would be much nicer if it could for example shrink entirely when there are no elements inside and expand with the increasing number of elements. The maximum size to which it could expand would be configurable.

**Comment by elParaguayo** [comment]
That's not quite right. It doesn't take a fixed amount of space. It's a `STRETCH` widget which means it expands to fill the available space. This would appear fixed if nothing else in your bar is changing its width. This is the same approach used for the `WindowName` widget.

What's the issue that this causes for you? Is it to do with background colour or decorations?
